backup/Duke.java

```diff
@@ -0,0 +1,116 @@
+import java.io.*;
+import java.util.Scanner;
+
+public class Duke {
+    private static final String FILENAME = "./data/duke.txt";
+    private static final File FILE = new File("./data/duke.txt");
+    private static final String LOGO = " ____        _        \n"
+                                     + "|  _ \\ _   _| | _____ \n"
+                                     + "| | | | | | | |/ / _ \\\n"
+                                     + "| |_| | |_| |   <  __/\n"
+                                     + "|____/ \\__,_|_|\\_\\___|\n";
+
+    // initialise the user and loads data from file to TaskList if file exists.
+    private static void initialise(Scanner scanner) throws FileNotFoundException {
+        String welcomeMsg = "Hello from\n" + LOGO;
+        welcomeMsg += Parser.parse("Hello, I'm Duke! What's your name?");
+        System.out.println(welcomeMsg);
+        String name = scanner.nextLine().strip();
+        if (!FILE.exists() && FILE.length() > 0) { // if file is empty or file does not exist
+            welcomeMsg = Parser.parse("Hello " + name + "! What can I do for you?");
+        } else { // initialise TaskList
+            BufferedReader br = new BufferedReader(new FileReader(FILE));
+            String input;
+            try {
+                while ((input = br.readLine()) != null) {
+//                    System.out.println("hello?");
+                    String[] inputArr = input.split(" | ");
+                    switch (inputArr[0]) {
+                        case "T":
+                            TaskList.addTask(inputArr[0], inputArr[2], inputArr[1]);
+                            break;
+                        case "D":
+                            TaskList.addTask(inputArr[0], inputArr[2], inputArr[3], Integer.valueOf(inputArr[1]));
+                            break;
+                        case "E":
+                            TaskList.addTask(inputArr[0], inputArr[2], inputArr[3], Integer.valueOf(inputArr[1]));
+                            break;
+                        default:
+                            break;
+                    }
+                }
+            } catch (IOException e) {
+
+            }
+            welcomeMsg = Parser.parse("Hello " + name + "! Here are your tasks:\n" + TaskList.list());
+        }
+        System.out.println(welcomeMsg);
+
+    }
+
+    private static void echo(String input) {
+        System.out.println(Parser.parse(input));
+    }
+
+    public static void main(String[] args) throws IOException {
+        //BufferedWriter writer = new BufferedWriter(new FileWriter(FILENAME));
+        Scanner scanner = new Scanner(System.in);
```

Comment:
A level of abstraction could be applied here. The Duke class can be referred from the week 3 project “AMoreOOP”, where a few lines will do, and the string parsing / logic handling can be handled by the Parser class. One reason being that outside viewers can easily break the program by altering the main class, such as editing the string parsing or trimming where all the business logic is located at.